### DISCUSSION TITLE
Install tool in Linux

### DISCUSSION THREAD

**Discussion by smedir** [RhetTbull/osxphotos#1602] [Q&A] [answered]
I see that the tool should run on Linux. I am trying to install the tool with the instructions provided. pip and pipx packages are for Darwin and will not install and give an erorr. I created a venv and could not complete install from a cloned git repository with the given instructions with similar errors-

```pip._vendor.packaging.version.InvalidVersion: Invalid version:```

I do not have a mac anymore, and the photos library is rsynced/archived on a zfs backup. I would appreciate any help  installing on linux command line.

**Comment by RhetTbull** [comment]
It used to run in linux. Sometime in last month, it stopped working but even old versions that used to work no longer work. I believe the issue has something to do with pipx or pip but I've not had time to figure out what is going on. I noticed this recently with the CI workflows that tested OSXPhotos on Ubuntu stopped working.

**Reply by RhetTbull** [reply]
I opened #1604 for this.

**Comment by RhetTbull** [comment, marked as answer]
Try this:

```bash
python3 -m venv venv-osxphotos
source venv-osxphotos/bin/activate
python3 -m pip install osxphotos
```

This worked for me on Ubuntu. Something about running from the venv appears to let this work. In the mean time I need to figure out what's actually causing this (it's somewhere down the dependency chain)

**Reply by smedir** [reply]
Thanks! that works. Now to dig out all the photos from the backup!

Thank you for this wonderful tool. It looks awesome.
